### PR TITLE
added profile 259 and fixed profile 47 folder name.

### PR DESCRIPTION
**Pull request by matthid**
I took the folder names from: http://embed.plnkr.co/03ck2dCtnJogBKHJ9EjY/preview

For profile259 to work some files which need Thread are `#if`d out. I'm not sure if the same functionality can be implemented with Tasks on profile259. I locked FSharp.Formatting for now as the current nuget package doesn't work on windows.

This also adds the FSharp.Core nuget package as there are problems with the default versions.

### CODE REVIEW

File: paket.dependencies [hunk 1]
```diff
@@ -2,8 +2,9 @@ source http://nuget.org/api/v2
 
 nuget FAKE
 nuget NUnit.Runners
-nuget FSharp.Formatting
+nuget FSharp.Formatting == 2.6.3
```
Comment (forki) [inline paket.dependencies]: why did you fix this?

Comment (matthid) [inline paket.dependencies]: > I locked FSharp.Formatting for now as the current nuget package doesn't work on windows.
See also https://github.com/tpetricek/FSharp.Formatting/issues/266

**Comment by matthid** [comment]
I hate this xbuild bug, is there a way to define multiple symbols from within FAKE that works on both platforms?

**Comment by forki** [comment]
it doesn't build on my machine

image

**Comment by matthid** [comment]
It builds on my machine, it doesn't build on travis because it seems to be non-trivial to define multiple symbols on travis, any ideas?
For your build failure, can you say which build is breaking? (47, 259 or net35?)

**Comment by matthid** [comment]
Appveyor can build as well: https://ci.appveyor.com/project/fsgit/fsharpx-collections/build/0.0.1.26

**Comment by forki** [comment]
I think it's 259, but after 2da8a94 I have a new one 47:

image

**Comment by matthid** [comment]
Yeah I had this problem before but I cannot remember how I solved it (of if I solved it at all). I thought with a space it would work on travis but it doesn't :(. The problem is to define multiple defines on xbuild...

**Comment by forki** [comment]
I'm on windows.

**Comment by matthid** [comment]
yep apparently it doesn't work on any system with a space :(

**Comment by matthid** [comment]
I removed the commit.

**Comment by matthid** [comment]
Now it "should" work on windows (it works on Appveyor and my machine). You should get your initial error, which I have no idea about.

**Comment by forki** [comment]
If you add reference to the FSharp.Core package you need to remove the old library reference see tohttps://github.com/matthid/FSharpx.Collections/blob/profile259/src/FSharpx.Collections.Experimental/FSharpx.Collections.Experimental.fsproj#L89

`paket update --hard` can do this for you

**Comment by forki** [comment]
You also need to list the new dependency in the nuget task in the build.fsx:

    Dependencies = 
        ["FSharp.Core", GetPackageVersion "packages" "FSharp.Core" ] })

**Comment by matthid** [comment]
I thought that's why we have `Condition="'$(TargetFrameworkProfile)' == ''"` in there to keep the reference for the non-portable build?

For the nuget package I'm not sure if its actually required by the consumer (because we can assume there is a FSharp.Core available on the target system?). But again I have no idea when we need it when we don't. Is this nuget FSharp.Core something special?

**Comment by matthid** [comment]
Ah, now I understand that the FSharp.Core nuget package has already covered everything, so I will just remove that reference and use the paket generated ones... I'm wondering why it works on my machine?

**Comment by forki** [comment]
FSharp.Core issues are the biggest pain in the whole ecosystem. Mostly because MS decided to not bundle it with .NET. Some have it installed - some don't. The package was created to make things easier. But everybody needs to use it. Not listing it as a dependency but compiling against it will only imcrease the pain level.

**Comment by matthid** [comment]
Yeah :( . Thanks for pointing that out!
But now I get your first error after removing the references oO.

**Comment by matthid** [comment]
I think the problem may be that the nuget FSharp.Core package is not compatible with net35...

**Comment by forki** [comment]
I think we don't need to keep support for net35. People can use the 1.9.6

**Comment by matthid** [comment]
Ok I found no xbuild command line that would define multiple symbols...
If you have no other idea we can only embed the defines into the fsproj which is less than ideal...
(within a `<When Condition,,, </Otherwise>`)

**Comment by matthid** [comment]
Ok I found one way:
Defining the Environment variable `DefineConstants="FX_PORTABLE;FX_NO_THREAD"` works, we could add that workaround to FAKE until xbuild supports that. But apparently a written
`<DefineConstants>...</DefineConstants>` is preferred over environment variables, so it makes sense to include the `/p:DefineConstants` parameter anyway on windows (because it works there). So the workaround doesn't work in all cases :(. Maybe include a warning in FAKE? Would you accept such a change or should I open a issue on FAKE to discuss it first?

That patch took waaaaay longer than expected. But its finally building :). Is this workaround acceptable?

**Comment by forki** [comment]
mhm I'm not sure I understand the change, but I guess I'd trust you on this ;-)

**Comment by matthid** [comment]
The problem is basically that 
`"/p:DefineConstants=FX_PORTABLE;FX_NO_THREAD"` is not working on xbuild :/ (xbuild will exit with an error message). And the workaround is to use the same called environment variable instead (At first I didn't even know that this is possible so I suggested to modify the msbuild files directly).

**Comment by matthid** [comment]
I still did not add the FSharp.Core nuget package to the dependencies by the way, can you do that yourself or should I open another pull request?

**Comment by forki** [comment]
I already did that. and released as 1.10.0

2015-02-26 19:41 GMT+01:00 matthid <notifications@github.com>:

**Comment by matthid** [comment]
Ah sorry didn't saw you already added it :+1:  , Thanks for the release!

**Pull request merged**